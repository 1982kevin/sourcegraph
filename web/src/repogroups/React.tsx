import { RepogroupMetadata } from './types'
import { SearchPatternType } from '../../../shared/src/graphql/schema'
import * as React from 'react'
export const react: RepogroupMetadata = {
    title: 'React',
    name: 'react',
    url: '/react',
    description:
        "Explore React repositories on GitHub. The search examples show how to use React Hook useState with various data types.",
    examples: [
        {
            title: 'Find imports of useState with regex search',
            exampleQuery: <>import [^;]+useState[^;]+ from 'react'</>,
            rawQuery: "import [^;]+useState[^;]+ from 'react'",
            patternType: SearchPatternType.regexp,
        },
        {
            title: 'Examples of useState with an object as the input parameter',
            description: `useState takes a single argument and can accept a primitive, an array or an object.
            Syntax for usage is [state_value, function_to_update_state_value] = useState (initial_state_value).`,
            exampleQuery: <>{'useState({:[string]})'}</>,
            rawQuery: 'useState({:[string]}) count:1000',
            patternType: SearchPatternType.structural,
        },
        {
            title: 'Examples of useState with any JavaScript data type as the input parameter',
            exampleQuery: <>useState(:[string])</>,
            rawQuery: 'useState(:[string]) count:1000',
            patternType: SearchPatternType.structural,
        },
        {
            title: 'Examples of useState with any JavaScript data type as the input parameter in only TypeScript files',
            exampleQuery: (
                <>
                    useState(:[string])
                    <span className="search-keyword">lang:</span>typescript
                </>
            ),
            rawQuery: 'useState(:[string]) count:1000 lang:typescript',
            patternType: SearchPatternType.structural,
        },
    ],
    homepageDescription: 'Learn React with code examples.',
    homepageIcon: 'https://cdn4.iconfinder.com/data/icons/logos-3/600/React.js_logo-512.png',
}
