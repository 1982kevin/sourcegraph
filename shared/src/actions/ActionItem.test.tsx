import * as H from 'history'
import React from 'react'
import { createBarrier } from '../api/integration-test/testHelpers'
import { NOOP_TELEMETRY_SERVICE } from '../telemetry/telemetryService'
import { ActionItem } from './ActionItem'
import { NEVER } from 'rxjs'
import { mount } from 'enzyme'

jest.mock('mdi-react/OpenInNewIcon', () => 'OpenInNewIcon')

describe('ActionItem', () => {
    const NOOP_EXTENSIONS_CONTROLLER = { executeCommand: () => Promise.resolve(undefined) }
    const NOOP_PLATFORM_CONTEXT = { forceUpdateTooltip: () => undefined, settings: NEVER }
    const history = H.createMemoryHistory()

    test('non-actionItem variant', () => {
        const component = mount(
            <ActionItem
                action={{ id: 'c', command: 'c', title: 't', description: 'd', iconURL: 'u', category: 'g' }}
                telemetryService={NOOP_TELEMETRY_SERVICE}
                location={history.location}
                extensionsController={NOOP_EXTENSIONS_CONTROLLER}
                platformContext={NOOP_PLATFORM_CONTEXT}
            />
        )
        expect(component).toMatchSnapshot()
    })

    test('actionItem variant', () => {
        const component = mount(
            <ActionItem
                action={{ id: 'c', command: 'c', title: 't', description: 'd', iconURL: 'u', category: 'g' }}
                telemetryService={NOOP_TELEMETRY_SERVICE}
                variant="actionItem"
                location={history.location}
                extensionsController={NOOP_EXTENSIONS_CONTROLLER}
                platformContext={NOOP_PLATFORM_CONTEXT}
            />
        )
        expect(component).toMatchSnapshot()
    })

    test('noop command', () => {
        const component = mount(
            <ActionItem
                action={{ id: 'c', title: 't', description: 'd', iconURL: 'u', category: 'g' }}
                telemetryService={NOOP_TELEMETRY_SERVICE}
                location={history.location}
                extensionsController={NOOP_EXTENSIONS_CONTROLLER}
                platformContext={NOOP_PLATFORM_CONTEXT}
            />
        )
        expect(component).toMatchSnapshot()
    })

    test('pressed toggle actionItem', () => {
        const component = mount(
            <ActionItem
                action={{ id: 'a', command: 'c', actionItem: { pressed: true, label: 'b' } }}
                telemetryService={NOOP_TELEMETRY_SERVICE}
                variant="actionItem"
                location={history.location}
                extensionsController={NOOP_EXTENSIONS_CONTROLLER}
                platformContext={NOOP_PLATFORM_CONTEXT}
            />
        )
        expect(component).toMatchSnapshot()
    })

    test('non-pressed actionItem', () => {
        const component = mount(
            <ActionItem
                action={{ id: 'a', command: 'c', actionItem: { pressed: false, label: 'b' } }}
                telemetryService={NOOP_TELEMETRY_SERVICE}
                variant="actionItem"
                location={history.location}
                extensionsController={NOOP_EXTENSIONS_CONTROLLER}
                platformContext={NOOP_PLATFORM_CONTEXT}
            />
        )
        expect(component).toMatchSnapshot()
    })

    test('title element', () => {
        const component = mount(
            <ActionItem
                action={{ id: 'c', command: 'c', title: 't', description: 'd', iconURL: 'u', category: 'g' }}
                telemetryService={NOOP_TELEMETRY_SERVICE}
                variant="actionItem"
                title={<span>t2</span>}
                location={history.location}
                extensionsController={NOOP_EXTENSIONS_CONTROLLER}
                platformContext={NOOP_PLATFORM_CONTEXT}
            />
        )
        expect(component).toMatchSnapshot()
    })

    test('run command', async () => {
        const { wait, done } = createBarrier()

        const component = mount(
            <ActionItem
                action={{ id: 'c', command: 'c', title: 't', description: 'd', iconURL: 'u', category: 'g' }}
                telemetryService={NOOP_TELEMETRY_SERVICE}
                variant="actionItem"
                disabledDuringExecution={true}
                location={history.location}
                extensionsController={{ ...NOOP_EXTENSIONS_CONTROLLER, executeCommand: () => wait }}
                platformContext={NOOP_PLATFORM_CONTEXT}
            />
        )
        component.simulate('click')
        expect(component).toMatchSnapshot()

        // Finish execution. (Use setTimeout to wait for the executeCommand resolution to result in the setState
        // call.)
        done()
        await new Promise<void>(resolve => setTimeout(resolve))
        expect(component).toMatchSnapshot()
    })

    test('run command with showLoadingSpinnerDuringExecution', async () => {
        const { wait, done } = createBarrier()

        const component = mount(
            <ActionItem
                action={{ id: 'c', command: 'c', title: 't', description: 'd', iconURL: 'u', category: 'g' }}
                telemetryService={NOOP_TELEMETRY_SERVICE}
                variant="actionItem"
                showLoadingSpinnerDuringExecution={true}
                location={history.location}
                extensionsController={{ ...NOOP_EXTENSIONS_CONTROLLER, executeCommand: () => wait }}
                platformContext={NOOP_PLATFORM_CONTEXT}
            />
        )

        component.simulate('click')
        expect(component).toMatchSnapshot()

        // Finish execution. (Use setTimeout to wait for the executeCommand resolution to result in the setState
        // call.)
        done()
        await new Promise<void>(resolve => setTimeout(resolve))
        expect(component).toMatchSnapshot()
    })

    test('run command with error', async () => {
        const component = mount(
            <ActionItem
                action={{ id: 'c', command: 'c', title: 't', description: 'd', iconURL: 'u', category: 'g' }}
                telemetryService={NOOP_TELEMETRY_SERVICE}
                variant="actionItem"
                disabledDuringExecution={true}
                location={history.location}
                extensionsController={{
                    ...NOOP_EXTENSIONS_CONTROLLER,
                    executeCommand: () => Promise.reject(new Error('x')),
                }}
                platformContext={NOOP_PLATFORM_CONTEXT}
            />
        )

        // Run command (which will reject with an error). (Use setTimeout to wait for the executeCommand resolution
        // to result in the setState call.)
        component.simulate('click')
        await new Promise<void>(resolve => setTimeout(resolve))
        expect(component).toMatchSnapshot()
    })

    test('run command with error with showInlineError', async () => {
        const component = mount(
            <ActionItem
                action={{ id: 'c', command: 'c', title: 't', description: 'd', iconURL: 'u', category: 'g' }}
                telemetryService={NOOP_TELEMETRY_SERVICE}
                variant="actionItem"
                showInlineError={true}
                location={history.location}
                extensionsController={{
                    ...NOOP_EXTENSIONS_CONTROLLER,
                    executeCommand: () => Promise.reject(new Error('x')),
                }}
                platformContext={NOOP_PLATFORM_CONTEXT}
            />
        )

        // Run command (which will reject with an error). (Use setTimeout to wait for the executeCommand resolution
        // to result in the setState call.)
        component.simulate('click')
        await new Promise<void>(resolve => setTimeout(resolve))
        expect(component).toMatchSnapshot()
    })

    describe('"open" command', () => {
        it('renders as link', () => {
            jsdom.reconfigure({ url: 'https://example.com/foo' })

            const component = mount(
                <ActionItem
                    action={{ id: 'c', command: 'open', commandArguments: ['https://example.com/bar'], title: 't' }}
                    telemetryService={NOOP_TELEMETRY_SERVICE}
                    location={history.location}
                    extensionsController={NOOP_EXTENSIONS_CONTROLLER}
                    platformContext={NOOP_PLATFORM_CONTEXT}
                />
            )
            expect(component).toMatchSnapshot()
        })

        it('renders as link with icon and opens a new tab for a different origin', () => {
            jsdom.reconfigure({ url: 'https://example.com/foo' })

            const component = mount(
                <ActionItem
                    action={{ id: 'c', command: 'open', commandArguments: ['https://other.com/foo'], title: 't' }}
                    telemetryService={NOOP_TELEMETRY_SERVICE}
                    location={history.location}
                    extensionsController={NOOP_EXTENSIONS_CONTROLLER}
                    platformContext={NOOP_PLATFORM_CONTEXT}
                />
            )
            expect(component).toMatchSnapshot()
        })

        it('renders as link that opens in a new tab, but without icon for a different origin as the alt action and a primary action defined', () => {
            jsdom.reconfigure({ url: 'https://example.com/foo' })

            const component = mount(
                <ActionItem
                    action={{ id: 'c1', command: 'whatever', title: 'primary' }}
                    altAction={{ id: 'c2', command: 'open', commandArguments: ['https://other.com/foo'], title: 'alt' }}
                    telemetryService={NOOP_TELEMETRY_SERVICE}
                    location={history.location}
                    extensionsController={NOOP_EXTENSIONS_CONTROLLER}
                    platformContext={NOOP_PLATFORM_CONTEXT}
                />
            )
            expect(component).toMatchSnapshot()
        })
    })
})
