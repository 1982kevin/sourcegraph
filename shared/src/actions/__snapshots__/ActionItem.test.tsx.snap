// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionItem "open" command renders as link 1`] = `
<ActionItem
  action={
    Object {
      "command": "open",
      "commandArguments": Array [
        "https://example.com/bar",
      ],
      "id": "c",
      "title": "t",
    }
  }
  extensionsController={
    Object {
      "executeCommand": [Function],
    }
  }
  location="[Location path=/]"
  platformContext={
    Object {
      "forceUpdateTooltip": [Function],
      "settings": "[Observable]",
    }
  }
  telemetryService={
    Object {
      "log": [Function],
    }
  }
>
  <LinkOrButton
    className="action-item"
    onSelect={[Function]}
    to="https://example.com/bar"
  >
    <AnchorLink
      className="action-item"
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={0}
      to="https://example.com/bar"
    >
      <a
        className="action-item"
        href="https://example.com/bar"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
         
        t
         
      </a>
    </AnchorLink>
  </LinkOrButton>
</ActionItem>
`;

exports[`ActionItem "open" command renders as link that opens in a new tab, but without icon for a different origin as the alt action and a primary action defined 1`] = `
<ActionItem
  action={
    Object {
      "command": "whatever",
      "id": "c1",
      "title": "primary",
    }
  }
  altAction={
    Object {
      "command": "open",
      "commandArguments": Array [
        "https://other.com/foo",
      ],
      "id": "c2",
      "title": "alt",
    }
  }
  extensionsController={
    Object {
      "executeCommand": [Function],
    }
  }
  location="[Location path=/]"
  platformContext={
    Object {
      "forceUpdateTooltip": [Function],
      "settings": "[Observable]",
    }
  }
  telemetryService={
    Object {
      "log": [Function],
    }
  }
>
  <LinkOrButton
    className="action-item"
    onSelect={[Function]}
    rel="noopener noreferrer"
    target="_blank"
    to="https://other.com/foo"
  >
    <AnchorLink
      className="action-item"
      onClick={[Function]}
      onKeyPress={[Function]}
      rel="noopener noreferrer"
      tabIndex={0}
      target="_blank"
      to="https://other.com/foo"
    >
      <a
        className="action-item"
        href="https://other.com/foo"
        onClick={[Function]}
        onKeyPress={[Function]}
        rel="noopener noreferrer"
        tabIndex={0}
        target="_blank"
      >
         
        primary
         
      </a>
    </AnchorLink>
  </LinkOrButton>
</ActionItem>
`;

exports[`ActionItem "open" command renders as link with icon and opens a new tab for a different origin 1`] = `
<ActionItem
  action={
    Object {
      "command": "open",
      "commandArguments": Array [
        "https://other.com/foo",
      ],
      "id": "c",
      "title": "t",
    }
  }
  extensionsController={
    Object {
      "executeCommand": [Function],
    }
  }
  location="[Location path=/]"
  platformContext={
    Object {
      "forceUpdateTooltip": [Function],
      "settings": "[Observable]",
    }
  }
  telemetryService={
    Object {
      "log": [Function],
    }
  }
>
  <LinkOrButton
    className="action-item"
    onSelect={[Function]}
    rel="noopener noreferrer"
    target="_blank"
    to="https://other.com/foo"
  >
    <AnchorLink
      className="action-item"
      onClick={[Function]}
      onKeyPress={[Function]}
      rel="noopener noreferrer"
      tabIndex={0}
      target="_blank"
      to="https://other.com/foo"
    >
      <a
        className="action-item"
        href="https://other.com/foo"
        onClick={[Function]}
        onKeyPress={[Function]}
        rel="noopener noreferrer"
        tabIndex={0}
        target="_blank"
      >
         
        t
         
        <OpenInNewIcon />
      </a>
    </AnchorLink>
  </LinkOrButton>
</ActionItem>
`;

exports[`ActionItem actionItem variant 1`] = `
<ActionItem
  action={
    Object {
      "category": "g",
      "command": "c",
      "description": "d",
      "iconURL": "u",
      "id": "c",
      "title": "t",
    }
  }
  extensionsController={
    Object {
      "executeCommand": [Function],
    }
  }
  location="[Location path=/]"
  platformContext={
    Object {
      "forceUpdateTooltip": [Function],
      "settings": "[Observable]",
    }
  }
  telemetryService={
    Object {
      "log": [Function],
    }
  }
  variant="actionItem"
>
  <LinkOrButton
    className="action-item action-item--variant-action-item"
    data-tooltip="d"
    onSelect={[Function]}
  >
    <a
      aria-label="d"
      className="action-item action-item--variant-action-item"
      data-tooltip="d"
      href=""
      onAuxClick={[Function]}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={0}
    >
      <img
        src="u"
      />
       
      g: 
      t
       
    </a>
  </LinkOrButton>
</ActionItem>
`;

exports[`ActionItem non-actionItem variant 1`] = `
<ActionItem
  action={
    Object {
      "category": "g",
      "command": "c",
      "description": "d",
      "iconURL": "u",
      "id": "c",
      "title": "t",
    }
  }
  extensionsController={
    Object {
      "executeCommand": [Function],
    }
  }
  location="[Location path=/]"
  platformContext={
    Object {
      "forceUpdateTooltip": [Function],
      "settings": "[Observable]",
    }
  }
  telemetryService={
    Object {
      "log": [Function],
    }
  }
>
  <LinkOrButton
    className="action-item"
    data-tooltip="d"
    onSelect={[Function]}
  >
    <a
      aria-label="d"
      className="action-item"
      data-tooltip="d"
      href=""
      onAuxClick={[Function]}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={0}
    >
      <img
        src="u"
      />
       
      g: 
      t
       
    </a>
  </LinkOrButton>
</ActionItem>
`;

exports[`ActionItem non-pressed actionItem 1`] = `
<ActionItem
  action={
    Object {
      "actionItem": Object {
        "label": "b",
        "pressed": false,
      },
      "command": "c",
      "id": "a",
    }
  }
  extensionsController={
    Object {
      "executeCommand": [Function],
    }
  }
  location="[Location path=/]"
  platformContext={
    Object {
      "forceUpdateTooltip": [Function],
      "settings": "[Observable]",
    }
  }
  telemetryService={
    Object {
      "log": [Function],
    }
  }
  variant="actionItem"
>
  <LinkOrButton
    className="action-item action-item--variant-action-item"
    onSelect={[Function]}
    pressed={false}
  >
    <a
      aria-pressed={false}
      className="action-item action-item--variant-action-item"
      href=""
      onAuxClick={[Function]}
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
       
      b
       
    </a>
  </LinkOrButton>
</ActionItem>
`;

exports[`ActionItem noop command 1`] = `
<ActionItem
  action={
    Object {
      "category": "g",
      "description": "d",
      "iconURL": "u",
      "id": "c",
      "title": "t",
    }
  }
  extensionsController={
    Object {
      "executeCommand": [Function],
    }
  }
  location="[Location path=/]"
  platformContext={
    Object {
      "forceUpdateTooltip": [Function],
      "settings": "[Observable]",
    }
  }
  telemetryService={
    Object {
      "log": [Function],
    }
  }
>
  <span
    className="action-item  "
    data-tooltip="d"
  >
    <img
      src="u"
    />
     
    g: 
    t
  </span>
</ActionItem>
`;

exports[`ActionItem pressed toggle actionItem 1`] = `
<ActionItem
  action={
    Object {
      "actionItem": Object {
        "label": "b",
        "pressed": true,
      },
      "command": "c",
      "id": "a",
    }
  }
  extensionsController={
    Object {
      "executeCommand": [Function],
    }
  }
  location="[Location path=/]"
  platformContext={
    Object {
      "forceUpdateTooltip": [Function],
      "settings": "[Observable]",
    }
  }
  telemetryService={
    Object {
      "log": [Function],
    }
  }
  variant="actionItem"
>
  <LinkOrButton
    className="action-item action-item--variant-action-item action-item--pressed"
    onSelect={[Function]}
    pressed={true}
  >
    <a
      aria-pressed={true}
      className="action-item action-item--variant-action-item action-item--pressed"
      href=""
      onAuxClick={[Function]}
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
       
      b
       
    </a>
  </LinkOrButton>
</ActionItem>
`;

exports[`ActionItem run command 1`] = `
<ActionItem
  action={
    Object {
      "category": "g",
      "command": "c",
      "description": "d",
      "iconURL": "u",
      "id": "c",
      "title": "t",
    }
  }
  disabledDuringExecution={true}
  extensionsController={
    Object {
      "executeCommand": [Function],
    }
  }
  location="[Location path=/]"
  platformContext={
    Object {
      "forceUpdateTooltip": [Function],
      "settings": "[Observable]",
    }
  }
  telemetryService={
    Object {
      "log": [Function],
    }
  }
  variant="actionItem"
>
  <LinkOrButton
    className="action-item action-item--variant-action-item"
    data-tooltip="d"
    disabled={true}
    onSelect={[Function]}
  >
    <a
      aria-label="d"
      className="action-item action-item--variant-action-item disabled"
      data-tooltip="d"
      href=""
      onAuxClick={[Function]}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={0}
    >
      <img
        src="u"
      />
       
      g: 
      t
       
    </a>
  </LinkOrButton>
</ActionItem>
`;

exports[`ActionItem run command 2`] = `
<ActionItem
  action={
    Object {
      "category": "g",
      "command": "c",
      "description": "d",
      "iconURL": "u",
      "id": "c",
      "title": "t",
    }
  }
  disabledDuringExecution={true}
  extensionsController={
    Object {
      "executeCommand": [Function],
    }
  }
  location="[Location path=/]"
  platformContext={
    Object {
      "forceUpdateTooltip": [Function],
      "settings": "[Observable]",
    }
  }
  telemetryService={
    Object {
      "log": [Function],
    }
  }
  variant="actionItem"
>
  <LinkOrButton
    className="action-item action-item--variant-action-item"
    data-tooltip="d"
    disabled={true}
    onSelect={[Function]}
  >
    <a
      aria-label="d"
      className="action-item action-item--variant-action-item disabled"
      data-tooltip="d"
      href=""
      onAuxClick={[Function]}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={0}
    >
      <img
        src="u"
      />
       
      g: 
      t
       
    </a>
  </LinkOrButton>
</ActionItem>
`;

exports[`ActionItem run command with error 1`] = `
<ActionItem
  action={
    Object {
      "category": "g",
      "command": "c",
      "description": "d",
      "iconURL": "u",
      "id": "c",
      "title": "t",
    }
  }
  disabledDuringExecution={true}
  extensionsController={
    Object {
      "executeCommand": [Function],
    }
  }
  location="[Location path=/]"
  platformContext={
    Object {
      "forceUpdateTooltip": [Function],
      "settings": "[Observable]",
    }
  }
  telemetryService={
    Object {
      "log": [Function],
    }
  }
  variant="actionItem"
>
  <LinkOrButton
    className="action-item action-item--variant-action-item"
    data-tooltip="d"
    disabled={true}
    onSelect={[Function]}
  >
    <a
      aria-label="d"
      className="action-item action-item--variant-action-item disabled"
      data-tooltip="d"
      href=""
      onAuxClick={[Function]}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={0}
    >
      <img
        src="u"
      />
       
      g: 
      t
       
    </a>
  </LinkOrButton>
</ActionItem>
`;

exports[`ActionItem run command with error with showInlineError 1`] = `
<ActionItem
  action={
    Object {
      "category": "g",
      "command": "c",
      "description": "d",
      "iconURL": "u",
      "id": "c",
      "title": "t",
    }
  }
  extensionsController={
    Object {
      "executeCommand": [Function],
    }
  }
  location="[Location path=/]"
  platformContext={
    Object {
      "forceUpdateTooltip": [Function],
      "settings": "[Observable]",
    }
  }
  showInlineError={true}
  telemetryService={
    Object {
      "log": [Function],
    }
  }
  variant="actionItem"
>
  <LinkOrButton
    className="action-item action-item--variant-action-item"
    data-tooltip="d"
    onSelect={[Function]}
  >
    <a
      aria-label="d"
      className="action-item action-item--variant-action-item"
      data-tooltip="d"
      href=""
      onAuxClick={[Function]}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={0}
    >
      <img
        src="u"
      />
       
      g: 
      t
       
    </a>
  </LinkOrButton>
</ActionItem>
`;

exports[`ActionItem run command with showLoadingSpinnerDuringExecution 1`] = `
<ActionItem
  action={
    Object {
      "category": "g",
      "command": "c",
      "description": "d",
      "iconURL": "u",
      "id": "c",
      "title": "t",
    }
  }
  extensionsController={
    Object {
      "executeCommand": [Function],
    }
  }
  location="[Location path=/]"
  platformContext={
    Object {
      "forceUpdateTooltip": [Function],
      "settings": "[Observable]",
    }
  }
  showLoadingSpinnerDuringExecution={true}
  telemetryService={
    Object {
      "log": [Function],
    }
  }
  variant="actionItem"
>
  <LinkOrButton
    className="action-item action-item--loading action-item--variant-action-item"
    data-tooltip="d"
    disabled={true}
    onSelect={[Function]}
  >
    <a
      aria-label="d"
      className="action-item action-item--loading action-item--variant-action-item disabled"
      data-tooltip="d"
      href=""
      onAuxClick={[Function]}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={0}
    >
      <img
        src="u"
      />
       
      g: 
      t
       
      <div
        className="action-item__loader"
      >
        <LoadingSpinner>
          <div
            className="loading-spinner "
          />
        </LoadingSpinner>
      </div>
    </a>
  </LinkOrButton>
</ActionItem>
`;

exports[`ActionItem run command with showLoadingSpinnerDuringExecution 2`] = `
<ActionItem
  action={
    Object {
      "category": "g",
      "command": "c",
      "description": "d",
      "iconURL": "u",
      "id": "c",
      "title": "t",
    }
  }
  extensionsController={
    Object {
      "executeCommand": [Function],
    }
  }
  location="[Location path=/]"
  platformContext={
    Object {
      "forceUpdateTooltip": [Function],
      "settings": "[Observable]",
    }
  }
  showLoadingSpinnerDuringExecution={true}
  telemetryService={
    Object {
      "log": [Function],
    }
  }
  variant="actionItem"
>
  <LinkOrButton
    className="action-item action-item--loading action-item--variant-action-item"
    data-tooltip="d"
    disabled={true}
    onSelect={[Function]}
  >
    <a
      aria-label="d"
      className="action-item action-item--loading action-item--variant-action-item disabled"
      data-tooltip="d"
      href=""
      onAuxClick={[Function]}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={0}
    >
      <img
        src="u"
      />
       
      g: 
      t
       
      <div
        className="action-item__loader"
      >
        <LoadingSpinner>
          <div
            className="loading-spinner "
          />
        </LoadingSpinner>
      </div>
    </a>
  </LinkOrButton>
</ActionItem>
`;

exports[`ActionItem title element 1`] = `
<ActionItem
  action={
    Object {
      "category": "g",
      "command": "c",
      "description": "d",
      "iconURL": "u",
      "id": "c",
      "title": "t",
    }
  }
  extensionsController={
    Object {
      "executeCommand": [Function],
    }
  }
  location="[Location path=/]"
  platformContext={
    Object {
      "forceUpdateTooltip": [Function],
      "settings": "[Observable]",
    }
  }
  telemetryService={
    Object {
      "log": [Function],
    }
  }
  title={
    <span>
      t2
    </span>
  }
  variant="actionItem"
>
  <LinkOrButton
    className="action-item action-item--variant-action-item"
    data-tooltip="d"
    onSelect={[Function]}
  >
    <a
      aria-label="d"
      className="action-item action-item--variant-action-item"
      data-tooltip="d"
      href=""
      onAuxClick={[Function]}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={0}
    >
      <span>
        t2
      </span>
       
    </a>
  </LinkOrButton>
</ActionItem>
`;
